src/Microsoft.OpenApi/Writers/IOpenApiWriter.cs

```diff
@@ -72,5 +74,13 @@ public interface IOpenApiWriter
         /// Flush the writer.
         /// </summary>
         void Flush();
+
+        /// <summary>
+        /// Writes out existing examples in a mediatype object
+        /// </summary>
+        /// <param name="writer"></param>
+        /// <param name="example"></param>
+        /// <param name="version"></param>
+        void WriteV2Examples(IOpenApiWriter writer, OpenApiExample example, OpenApiSpecVersion version);
```

Comment:
this change is source breaking. I'm not sure what's been our policy around versioning in this lib, but if we follow semver (which seems like we do), this would mandate a major bump.

Reply:
This method also seems really out of place in this interface.  If I am creating a new writer, why would I have to create a special method for writing v2 examples.

Reply:
The original request from the team who need this primarily care about the fact that they can't read these examples. They don't have a use case for round-tripping them.  If we really need a breaking change then we can push the support for round tripping them to v2 and just ship the reading for V1.x

Reply:
I can refactor this by abstracting the method from the interface so that its no longer a breaking change. The round-tripping does not introduce a breaking change so I think we can retain that functionality.